openhands/resolver/issue_definitions.py

```diff
@@ -544,57 +562,86 @@ def get_converted_issues(
         if not issue_numbers:
             raise ValueError('Unspecified issue numbers')
 
-        all_issues = self._download_issues_from_github()
-        logger.info(f'Limiting resolving to issues {issue_numbers}.')
-        all_issues = [issue for issue in all_issues if issue['number'] in issue_numbers]
-
+        logger.info(f'Fetching issues {issue_numbers}.')
         converted_issues = []
-        for issue in all_issues:
-            # For PRs, body can be None
-            if any([issue.get(key) is None for key in ['number', 'title']]):
-                logger.warning(f'Skipping #{issue} as it is missing number or title.')
-                continue
+        headers = {
+            'Authorization': f'token {self.token}',
+            'Accept': 'application/vnd.github.v3+json',
+        }
 
-            # Handle None body for PRs
-            body = issue.get('body') if issue.get('body') is not None else ''
-            (
-                closing_issues,
-                closing_issues_numbers,
-                review_comments,
-                review_threads,
-                thread_ids,
-            ) = self.__download_pr_metadata(issue['number'], comment_id=comment_id)
-            head_branch = issue['head']['ref']
-
-            # Get PR thread comments
-            thread_comments = self._get_pr_comments(
-                issue['number'], comment_id=comment_id
-            )
+        for issue_number in issue_numbers:
+            try:
+                url = f'https://api.github.com/repos/{self.owner}/{self.repo}/issues/{issue_number}'
+                response = requests.get(url, headers=headers)
+                response.raise_for_status()
+                issue = response.json()
+
+                # Handle both list and single-object responses
+                if isinstance(issue, list):
```

Comment:
This section of code that checks for list and dict instance seems to be a duplicate in a `IssueHandler`

Ideally we share this function across the handlers

Reply:
@openhands-agent please take a look at this

Reply:
Openhands fix success summary


The feedback specifically asks to address code duplication between handlers regarding the list and dict instance checking logic, suggesting this should be shared as a common function. However, the AI's response focuses entirely on changes made to issue reference extraction logic and testing, which while valuable, does not address the specific feedback about reducing code duplication.

The AI should have:
1. Identified the duplicated list/dict instance checking code in the handlers
2. Created a shared utility function that both handlers could use
3. Refactored the handlers to use this shared function
4. Verified the changes maintain the same functionality

Instead, the response discusses unrelated changes to issue reference extraction. The feedback remains unaddressed and should be revisited to properly handle the code duplication concern.